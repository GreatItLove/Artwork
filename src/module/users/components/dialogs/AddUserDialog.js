import React from 'react';
import Dialog from 'material-ui/Dialog';
import PropTypes from 'prop-types';

import AddUser from '../../containers/AddUser';

const propTypes = {
  adding: PropTypes.bool.isRequired,
  closeAddDialog: PropTypes.func.isRequired
};

/**
 * @function AddUserDialog
 * @param {Boolean} adding
 * @param {Function} closeAddDialog
 * @returns {XML}
 */
function AddUserDialog({ adding, closeAddDialog }) {
  const customContentStyle = {
    width: '57%',
    position: 'fixed',
    top: '0',
    bottom: '0',
    left: '0',
    right: '0',
    maxWidth: 'none',
  };
  return (
    <Dialog
      onRequestClose={closeAddDialog}
      open={adding}
      bodyStyle={{ overflowY: 'unset' }}
      contentStyle={customContentStyle}
      overlayStyle={{ backgroundColor: '#797979', opacity: '0.8' }}
      paperProps={{ style: { backgroundColor: 'none' }, zDepth: 0 }}
    >
      {adding && <AddUser closeDialog={closeAddDialog} />}
    </Dialog>
  );
}

AddUserDialog.propTypes = propTypes;

export default AddUserDialog;
