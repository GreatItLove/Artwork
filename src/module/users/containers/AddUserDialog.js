import { connect } from 'react-redux';

import AddUserDialog from '../components/dialogs/AddUserDialog';
import { closeUserAddDialog } from '../usersActions';

const mapStateToProps = state => ({
  adding: state.users.adding
});

const mapDispatchToProps = dispatch => ({
  closeAddDialog: () => {
    dispatch(closeUserAddDialog());
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(AddUserDialog);
