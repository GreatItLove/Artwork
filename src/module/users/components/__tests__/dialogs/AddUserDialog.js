import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import AddUserDialog from '../../dialogs/AddUserDialog';

describe('AddUserDialog', () => {
  const props = {
    adding: true,
    closeAddDialog: jest.fn()
  };
  it('Snapshot of AddUserDialog when adding is true', () => {
    const wrapper = shallow(<AddUserDialog {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it('Snapshot of AddUserDialog whan adding is false ', () => {
    const propsAddingFalse = { ...props, ...{ adding: false } };
    const wrapper = shallow(<AddUserDialog {...propsAddingFalse} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
