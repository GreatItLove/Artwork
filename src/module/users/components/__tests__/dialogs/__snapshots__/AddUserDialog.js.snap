// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddUserDialog Snapshot of AddUserDialog whan adding is false  1`] = `
<Dialog
  autoDetectWindowHeight={true}
  autoScrollBodyContent={false}
  bodyStyle={
    Object {
      "overflowY": "unset",
    }
  }
  contentStyle={
    Object {
      "bottom": "0",
      "left": "0",
      "maxWidth": "none",
      "position": "fixed",
      "right": "0",
      "top": "0",
      "width": "57%",
    }
  }
  modal={false}
  onRequestClose={[Function]}
  open={false}
  overlayStyle={
    Object {
      "backgroundColor": "#797979",
      "opacity": "0.8",
    }
  }
  paperProps={
    Object {
      "style": Object {
        "backgroundColor": "none",
      },
      "zDepth": 0,
    }
  }
  repositionOnUpdate={true}
/>
`;

exports[`AddUserDialog Snapshot of AddUserDialog when adding is true 1`] = `
<Dialog
  autoDetectWindowHeight={true}
  autoScrollBodyContent={false}
  bodyStyle={
    Object {
      "overflowY": "unset",
    }
  }
  contentStyle={
    Object {
      "bottom": "0",
      "left": "0",
      "maxWidth": "none",
      "position": "fixed",
      "right": "0",
      "top": "0",
      "width": "57%",
    }
  }
  modal={false}
  onRequestClose={[Function]}
  open={true}
  overlayStyle={
    Object {
      "backgroundColor": "#797979",
      "opacity": "0.8",
    }
  }
  paperProps={
    Object {
      "style": Object {
        "backgroundColor": "none",
      },
      "zDepth": 0,
    }
  }
  repositionOnUpdate={true}
>
  <Translate(Connect(ReduxForm))
    closeDialog={[Function]}
  />
</Dialog>
`;
